iot-hub/Samples/service/CleanUpDevicesSample/CleanUpDevicesSample.cs

```diff
@@ -83,50 +50,52 @@ public async Task RunCleanUpAsync()
                     _devicesToDelete.RemoveRange(0, i);
```

Comment:
I question how well this will work. We're incrementing i, but also removing those items from the collection.

If I were to write this, I'd go from the end to the beginning, and remove as I go. That way the index of the next batch of items don't get impacted by removals.

Reply:
We are removing these items from the collection once we're done incrementing i and adding those items to another collection. One problem I see here is if the registry manage operation fails, we've removed the items from our primary list, but they haven't been deleted. So I am moving the remove operation after registry manager operation.